src/type-layout.md

```diff
@@ -162,7 +163,24 @@ representation will not change the layout of `Inner`.
 Nominal types without a `repr` attribute have the default representation.
 Informally, this representation is also called the `rust` representation.
 
-There are no guarantees of data layout made by this representation.
+There are very few data layout guarantees made by this representation. The only
+guarantees are:
+
+ 1. The fields of the struct are properly aligned.
+ 2. The fields do not overlap.
+ 3. The alignment of the struct is not less than the alignment of any of its
+    fields.
+
+Formally, the first guarantee means that the offset of any field in the struct
+is divisible by that field's alignment. The second guarantee means that the
+fields can be ordered such that the offset plus the size of any field is less
+than or equal to the offset of the next field in the ordering. The ordering does
+not have to be the same as the order in which the field are specified in the
+declaration of the struct.
```

Comment:
(also can be ignored)

If non-normative statements that provide motivation/context are permitted, it would also be worth noting here that only the compiler has enough information to optimally lay out a generic type for all possible type substitutions. i.e. `repr(C)` gets you consistency and control at the cost of sometimes not being able to always put generic fields in the best place.